GitUI/CommandsDialogs/CommitDialog/FormattedCommitMessage.cs

```diff
@@ -0,0 +1,343 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using GitCommands;
+
+namespace GitUI.CommandsDialogs.CommitDialog
+{
+    public class FormattedCommitMessage
+    {
+        public class ColoredLine
```

Comment:
Because it represents a colored line :smile: , when it is *not* wrapped (because it is the commit title, or because `AutoWrap` is not set). `From` is where the coloring begins.
I've been sparse on comment-doc because I intended all this as implementation detail, not really a component expected to be reused (it is public because it could eventually be, and anyway is a testable component).